test/test_tensordict.py

```diff
@@ -1757,6 +1757,25 @@ def test_split(self, td_name, device, performer):
                                 expected_inner_tensor_size
                             )
 
+    def test_pop(self, td_name, device):
+        td = getattr(self, td_name)(device)
+        assert 'a' in td
+        out = td.pop("a")
+        assert out == getattr(self, td_name)(device)['a']
+        assert 'a' not in td
+
+        default = 'some value'
+        out = td.pop("b", default)
+        assert out != default
+        out = td.pop("z", default)
+        assert out == default
+
+        with pytest.raises(
+            KeyError,
+            match = "You are trying to pop key z which is not in dict and not providing default value"
+        ):
+            td.pop("z")
```

Comment:
I agree this will run as expected but it's a bit tricky since we call pop with that key before (requires the reader to relate to the previous test, observe that the key wasn't there and expect this line to behave regardless)
Maybe using a distinct key would be more straightforward?

Reply:
Yep, done